src/py21cmfast/wrapper.py

```diff
@@ -2673,6 +2712,22 @@ def run_lightcone(
                 regenerate=regenerate,
             )
 
+            if coeval_callback is not None:
+                if compute_coeval_callback[iz]:
+                    coeval = Coeval(
+                        redshift=z,
+                        initial_conditions=init_box,
+                        perturbed_field=pf2,
+                        ionized_box=ib2,
+                        brightness_temp=bt2,
+                        ts_box=st2 if flag_options.USE_TS_FLUCT else None,
+                        photon_nonconservation_data=_get_photon_nonconservation_data()
+                        if flag_options.PHOTON_CONS
+                        else None,
+                        _globals=None,
+                    )
+                    coeval_callback_output.append(coeval_callback(coeval))
```

Comment:
You can wrap this in a Try: Except:, but in the Except, still raise an error -- it just means the error can maybe make a little more sense (you can attach which redshift it broke at, for example). Perhaps even better -- if it's the first time in the function that the callback is being called, and it fails, raise an error. Otherwise, raise a warning and keep going.